itest/lnd_sweep_test.go

```diff
@@ -1888,22 +1905,26 @@ func runBumpFee(ht *lntest.HarnessTest, alice *node.HarnessNode) {
 		SatPerVbyte: startFeeRate,
 		// The budget and the deadline delta must be set together.
 		Budget:        smallBudget,
-		DeadlineDelta: uint32(sweep.DefaultDeadlineDelta),
+		DeadlineDelta: 1,
 	}
 	alice.RPC.BumpFee(bumpFeeReq)
 
 	// Assert the pending sweep is created with the expected values:
 	// - broadcast attempts: 7.
-	// - starting fee rate: 1 sat/vbyte.
-	// - deadline: 1008.
+	// - starting fee rate: 0 sat/vbyte.
+	// - deadline: 1.
 	// - budget: 1000 sats.
+	//
+	// NOTE: The starting fee rate is 0 instead of 1 because the budget has
```

Comment:
I am a bit confused here, how can the starting feerate be 0, you mean not set at all and it will be set later in the fee_function ?

Reply:
I think it is explained below - the sweeper will update the starting fee rate to the last attempted fee rate, which in this case is 0, because we don't set it when error out in `ErrMaxPosistion`.

Reply:
ok decided to change the behavior. So previously two things were happening,
- we update the starting fee rate when there's `TxFailed` event to prepare for the next fee bump
- we set the next fee rate in the fee bump it when the budget is not enough or the inputs are not enough
This was why the itest was updated -this particular fee bump will fail with max position error, which means a) it's a failed event, so the fee rate will be updated and, b) it's a max position error, so the next fee rate was not set in the bump result, hence when we update the starting fee rate, it will be set to 0 (empty value).

This is now changed that, if there's a max position error, we always set the next fee rate before returning the fee bump result to the sweeper to handle.